core/server/services/settings/index.js

```diff
@@ -4,16 +4,78 @@
  */
 
 var SettingsModel = require('../../models/settings').Settings,
-    SettingsCache = require('./cache');
+    SettingsCache = require('./cache'),
+    SettingsLoader = require('./loader'),
+    _ = require('lodash'),
+    common = require('../../lib/common');
+    // SettingsFiles = require('./ensure-settings');
 
 module.exports = {
     init: function init() {
+        // TODO: the ensure settings fn up before the settings cache
+        // return SettingsFiles(this.knownSettings()).then(() => {
```

Comment:
I think you can already enable this code. This PR won't get merged before the CLI PR has been released.
Or did you had a different plan in mind?

> return SettingsFiles(this.knownSettings()).then(() => {

BTW: This won't work, because

1. will run into circular dependency problems - `ensure-settings` helper requires this file, this file requires `ensure-settings`
2. `ensure-settings` should just accept the known settings as argument